src/controllers/attachments/index.ts

```diff
@@ -1,32 +1,94 @@
-import { Controller, Get, Route } from 'tsoa'
+import { UploadedFile, Request, Controller, Get, Route, Path, Post, Header, Response, SuccessResponse } from 'tsoa'
 import { Logger } from 'pino'
+import express from 'express'
 
 import { logger } from '../../lib/logger'
-import Database from '../../lib/db'
+import Database, { Models, Query } from '../../lib/db'
 import type { Attachments } from '../../models'
+import { BadReqeust, NotFound } from '../../lib/error-handler'
+
+type File = {
+  [k: string]: string
+}
 
 @Route('attachments')
 export class attachments extends Controller {
   log: Logger
-  // TMP update once we have more defined schema
-  dbClient: any = new Database()
-  db: any
+  dbClient: Database = new Database()
+  db: Models<Query> | any // TMP this is the only one could not address now
 
   constructor() {
     super()
     this.log = logger.child({ controller: '/attachments' })
-    this.db = this.dbClient.db()
+    // TMP will be updated with a wrapper so ORM client independant
+    this.db = this.dbClient.init()
   }
 
   @Get('/')
-  public async get(): Promise<{ status: number; attachments: Attachments[] }> {
-    this.log.debug({
-      msg: 'this is a test route to validate tsoa/swagger',
+  @SuccessResponse(201, 'returns all attachments')
+  public async get(): Promise<Attachments[]> {
+    this.log.debug('retrieving all attachments')
+    const result = await this.db.attachments()
+
+    return result.map(({ binary_blob, ...item }: Attachments) => ({
+      ...item,
+      size: binary_blob.size,
+    }))
+  }
+
+  @Post('/')
+  @SuccessResponse(201, 'attachemnt has been created')
+  public async create(@Request() req: express.Request, @UploadedFile() file: File): Promise<string> {
+    this.log.debug(`crating an attachment ${JSON.stringify(file || req.body)}`)
+    if (file) {
+      await this.db.attachments().insert({
+        filename: file.originalname,
+        binary_blob: Buffer.from(file.buffer),
+      })
+    }
+
+    if (!req.body) throw new Error('nothing to upload') // TODO return correct (badreq)
+
+    await this.db.attachments().insert({
+      filename: 'json',
+      binary_blob: Buffer.from(JSON.stringify(req.body)),
     })
 
-    return {
-      status: 200,
-      attachments: await this.db.attachments(),
+    return 'success'
```

Comment:
we surely want it to return the `id`? Otherwise you can't reference the attachment you just created in a subsequent API call